plugins/view-resources/src/components/ViewOptionsButton.svelte

```diff
@@ -13,37 +13,31 @@
 // limitations under the License.
 -->
 <script lang="ts">
-  import { IssuesDateModificationPeriod, IssuesGrouping, IssuesOrdering, ViewOptions } from '@anticrm/tracker'
   import { Button, eventToHTMLElement, IconDownOutline, showPopup, Label } from '@anticrm/ui'
-  import { getViewOptions, setViewOptions } from '@anticrm/view-resources'
-  import view, { Viewlet } from '@anticrm/view'
-
+  import view from '@anticrm/view'
   import ViewOptionsPopup from './ViewOptionsPopup.svelte'
-  import { viewOptionsStore } from '../../viewOptions'
-
-  export let viewlet: Viewlet | undefined
+  import { getViewOptions, setViewOptions, viewOptionsStore, ViewOptionModel } from '../viewOptions'
 
-  let viewOptions: ViewOptions
-  $: if (viewlet) {
-    const savedViewOptions = getViewOptions(viewlet._id)
-    viewOptions = savedViewOptions
-      ? JSON.parse(savedViewOptions)
-      : {
-          groupBy: IssuesGrouping.Status,
-          orderBy: IssuesOrdering.Status,
-          completedIssuesPeriod: IssuesDateModificationPeriod.All,
-          shouldShowEmptyGroups: false,
-          shouldShowSubIssues: false
-        }
-  }
+  export let config: ViewOptionModel[]
+  export let viewOptionsKey: string
 
-  $: $viewOptionsStore = viewOptions
+  $: $viewOptionsStore = config.reduce(
```

Comment:
I'm confused, Svelte, will subscribe for viewOptionsStore and create local $viewOptionsStore, why you update it? if you plan to update store value, pleas use viewOptionsStore.set() instead.

Reply:
Is this misuse of store?
https://github.com/hcengineering/anticrm/blob/190954e74792b24263ddf340b9202180c3ef559d/plugins/view-resources/src/components/filter/FilterBar.svelte#L89

Reply:
It will update local variable only, not store.

Reply:
I already change it to update store explicit